### PR TITLE
implemented immutable class solution

### PR DESCRIPTION
**Pull request by MaliukDaria**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,13 +1,88 @@
...
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        List<Wheel> newWeelList = new ArrayList<>();
+        for (Wheel wheel : wheels) {
+            newWeelList.add(wheel.clone());
+        }
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Car.java:22]: let's move it to the separate private method

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,13 +1,88 @@
...
+        this.wheels = newWeelList;
+        if (engine != null) {
+            this.engine = engine.clone();
+        } else {
+            this.engine = null;
+        }
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Car.java:28]: the same about separate private method

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,13 +1,88 @@
...
+    public int getYear() {
+        return year;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public List<Wheel> getWheels() {
+        return new ArrayList<>(wheels);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Car.java:40]: do we need to clone all elements at the list?

Comment (MaliukDaria) [reply src/main/java/core/basesyntax/Car.java]: yes, done)

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (MaliukDaria) [reply src/main/java/core/basesyntax/Car.java]: done,  and added a null-check in cloneWheelList method (it would be good to add tests for null Wheel =) )

File: src/main/java/core/basesyntax/Car.java (hunk 2 above)
Comment (MaliukDaria) [reply src/main/java/core/basesyntax/Car.java]: done
